numpy/core/src/umath/override.c

```diff
@@ -576,19 +583,18 @@ PyUFunc_CheckOverride(PyUFuncObject *ufunc, char *method,
             break;
         }
     }
-
+    status = 0;
     /* Override found, return it. */
-    Py_XDECREF(method_name);
-    Py_XDECREF(normal_kwds);
-    Py_DECREF(override_args);
-    return 0;
-
+    goto cleanup;
```

Comment:
Kind of ugly. Why not get rid of `fail` and just set `status = -1; goto cleanup;`? Maybe replace `status` with `retval` or something else more informative.